### ISSUE TITLE
Error on probing IPv6-only host from an IPv4-only(?) container environment

### ISSUE THREAD

**Issue by Zash** [horazont/xmpp-blackbox-exporter#23]
To reproduce:

0. `podman build . -t docker.io/horazont/prometheus-xmpp-blackbox-exporter:latest`
1. `podman run -it --rm -p 9604:9604 docker.io/horazont/prometheus-xmpp-blackbox-exporter:latest`
2. `curl http://localhost:9604/probe?module=s2s_normal\&target=xmpp:ipv6-only.badxmpp.eu`

This fails with `curl` reporting an empty response and the running exporter prints the following traceback:

```
2021/09/30 18:14:49 http: panic serving 10.0.2.100:55498: runtime error: invalid memory address or nil pointer dereference
goroutine 39 [running]:
net/http.(*conn).serve.func1(0xc0001a4820)
	net/http/server.go:1795 +0x139
panic(0xa4bfa0, 0x10550f0)
	runtime/panic.go:679 +0x1b2
github.com/horazont/xmpp-blackbox-exporter/internal/prober.dialXMPP(0xbff4c0, 0xc00017a840, 0x106d100, 0xc00008f080, 0x0, 0x0, 0x0, 0x14, 0xc00016ef40, 0x14, ...)
	github.com/horazont/xmpp-blackbox-exporter@/internal/prober/dial.go:174 +0x1cd
github.com/horazont/xmpp-blackbox-exporter/internal/prober.ProbeS2S(0xbff4c0, 0xc00017a840, 0xc0001640b2, 0xa, 0xc0001640c4, 0x19, 0xc0001f21d0, 0x3, 0x6fc23ac00, 0x0, ...)
	github.com/horazont/xmpp-blackbox-exporter@/internal/prober/s2s_tls.go:139 +0xaa2
main.probeHandler(0xbfd640, 0xc0002900e0, 0xc0001f8200, 0xc000168650, 0xc00016cc60)
	github.com/horazont/xmpp-blackbox-exporter@/cmd/xmpp_blackbox_exporter/xmpp_blackbox_exporter.go:147 +0x6e4
main.run.func2(0xbfd640, 0xc0002900e0, 0xc0001f8200)
	github.com/horazont/xmpp-blackbox-exporter@/cmd/xmpp_blackbox_exporter/xmpp_blackbox_exporter.go:229 +0x84
net/http.HandlerFunc.ServeHTTP(0xb38438, 0xbfd640, 0xc0002900e0, 0xc0001f8200)
	net/http/server.go:2036 +0x44
net/http.(*ServeMux).ServeHTTP(0x106d040, 0xbfd640, 0xc0002900e0, 0xc0001f8200)
	net/http/server.go:2416 +0x1bd
net/http.serverHandler.ServeHTTP(0xc000290000, 0xbfd640, 0xc0002900e0, 0xc0001f8200)
	net/http/server.go:2831 +0xa4
net/http.(*conn).serve(0xc0001a4820, 0xbff440, 0xc0001fa4c0)
	net/http/server.go:1919 +0x875
created by net/http.(*Server).Serve
	net/http/server.go:2957 +0x384
```

Seems to work fine when run outside of a container, so it looks to me that it's related to the common lack of IPv6 support in containerization things. Running with `--network host` also seems to work, further pointing to container networking.

**Comment by horazont** [comment]
Does not seem to happen with current master. I get an error instead:

```json
{
  "level": "error",
  "ts": "2021-10-03T14:51:59.770Z",
  "caller": "prober/c2s_tls.go:135",
  "msg": "failed to dial for c2s probe",
  "module": "xmpp-client",
  "target": "xmpp:ipv6-only.badxmpp.eu",
  "err": "dial tcp [2001:41c9:1:424::35]:5222: connect: cannot assign requested address",
  "stacktrace": "github.com/horazont/xmpp-blackbox-exporter/internal/prober.ProbeC2S\n\tgithub.com/horazont/xmpp-blackbox-exporter/internal/prober/c2s_tls.go:135\nmain.probeHandler\n\tcommand-line-arguments/xmpp_blackbox_exporter.go:151\nmain.run.func2\n\tcommand-line-arguments/xmpp_blackbox_exporter.go:233\nnet/http.HandlerFunc.ServeHTTP\n\tnet/http/server.go:2049\nnet/http.(*ServeMux).ServeHTTP\n\tnet/http/server.go:2428\nnet/http.serverHandler.ServeHTTP\n\tnet/http/server.go:2867\nnet/http.(*conn).serve\n\tnet/http/server.go:1932"
}
```

I guess I need to cut a release.

**Comment by horazont** [comment]
Released 0.6.0, updated the image, should not crash now. Please report back if it does not work.
